src/client/datascience/raw-kernel/rawKernel.ts

```diff
@@ -111,7 +113,16 @@ export class RawKernel implements Kernel.IKernel {
         await this.kernelProcess.dispose();
         this.socket.dispose();
     }
-    public getSpec(): Promise<Kernel.ISpecModel> {
+    public async getSpec(): Promise<Kernel.ISpecModel> {
+        if (this.kernelProcess.kernelConnectionMetadata.kind === 'startUsingKernelSpec') {
+            const kernelSpec = cloneDeep(this.kernelProcess.kernelConnectionMetadata.kernelSpec) as any;
+            const resources = 'resources' in kernelSpec ? kernelSpec.resources : {};
+            return {
+                ...kernelSpec,
+                resources
+            };
+        }
+        traceError('Fetching kernel spec from raw kernel using JLab API');
         return this.realKernel.getSpec();
```

Comment:
This API doesn't work, `realKernel.getSpec` goes to Jupyter Lab API and that attempts to make a REST call to the base (dummy url).
This should not get executed hence added error logging.